enhancements/ingress/lb-allowed-source-ranges.md

```diff
@@ -0,0 +1,292 @@
+The current version of the document does not mention `.Spec.LoadBalancerSourceRanges` anymore.
+
+### Test Plan
+
+The operator that manages the IngressController Deployment and related
+resources has unit test coverage; for this enhancement, the unit tests are
+expanded to cover the additional functionality.  
+
+The operator has end-to-end tests; for this enhancement, a test is added that verifies
+the following scenarios:
+
+1. Adding/deleting/changing `AllowedSourceRanges` on the IngressController and 
+observing `.Spec.LoadBalancerSourceRanges` on the LoadBalancer Service.
+2. Adding `service.beta.kubernetes.io/load-balancer-source-ranges` annotation to 
+a LoadBalancer Service and observing cluster's upgradable status.
+3. Setting `.Spec.LoadBalancerSourceRanges` on the LoadBalancer Service 
+when `AllowedSourceRanges` is empty and observing cluster's upgradable status.
+
+This enhancement adds an upgrade test in openshift/origin to make sure that 
+the migration workflow explained under **Variation** above works seamlessly.
+In detail, it implements 
+[Test](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/upgrades/upgrade.go#L46) 
+interface that includes the following steps:
+
+1. In `Setup`, create an IngressController that requests an LB, 
+and set the `service.beta.kubernetes.io/load-balancer-source-ranges` annotation
+on the resulting LoadBalancer-type service.
+2. In `Test`, periodically check the service to verify that nothing updates the annotation
+or the `Spec.LoadBalancerSourceRanges` field on the service during the upgrade.
+3. In `Test`, verify again when the upgrade is done that nothing has updated the annotation
+or `Spec.LoadBalancerSourceRanges`, and verify that the operator has set `Upgradeable=False`.
+4. Then set `AllowedSourceRanges`, remove the annotation, and finally verify that the operator 
+has set `Upgradeable=True` and `Spec.LoadBalancerSourceRanges` accordingly.
+
+### Graduation Criteria
+
+N/A
+
+#### Dev Preview -> Tech Preview
+
+N/A. This feature will go directly to GA.
+
+#### Tech Preview -> GA
+
+N/A. This feature will go directly to GA.
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+The `AllowedSourceRanges` field is empty by default, so it will be empty on an upgraded cluster, until the cluster admin 
+sets it after the upgrade. If the cluster admin sets `AllowedSourceRanges` and then downgrades the cluster, 
+the `AllowedSourceRanges` field will vanish from the CRD; however, the downgraded operator would not update or 
+remove the `Spec.LoadBalancerSourceRanges` field value and would not update or remove the `service.beta.kubernetes.io/load-balancer-source-ranges`
+annotation. In any case, none of this (not `AllowedSourceRanges` on the IngressController, nor `LoadBalancerSourceRanges` or 
+`service.beta.kubernetes.io/load-balancer-source-ranges` on the service) should have any effect on any deployment.
```

Comment:
> In any case, none of this (not `AllowedSourceRanges` on the IngressController, nor `LoadBalancerSourceRanges` or 
`service.beta.kubernetes.io/load-balancer-source-ranges` on the service) should have any effect on any deployment.

I assume that this sentence was added in response to my earlier review comment: https://github.com/openshift/enhancements/pull/1177#discussion_r954471110.  When I made that comment, you had a statement in the EP that the deployment would be updated, and I only meant that that earlier statement in the EP should be modified or removed; I didn't mean that an explicit statement that the deployment wouldn't be updated was necessary.  Sorry for the confusion!

Reply:
Right, I deleted that sentence. No worries!